### PR TITLE
MISUV-2996 Update Calc output for R7 by adding National Insurance contributions table

### PR DESCRIPTION
**Pull request by andriikondiuk**

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -218,18 +218,23 @@ taxCal_breakdown.additional_charges                                 = Additional
...
+taxCal_breakdown.table.nic4.HRT                                     = Class 4 National Insurance ({0} at {1}%)
 taxCal_breakdown.table.ART                                          = Additional rate ({0} at {1}%)
 taxCal_breakdown.table.ART_scottish                                 = Top rate ({0} at {1}%)
 taxCal_breakdown.table.ZRTBR                                        = Basic rate band at nil rate ({0} at {1}%)
 taxCal_breakdown.table.ZRTHR                                        = Higher rate band at nil rate ({0} at {1}%)
 taxCal_breakdown.table.ZRTAR                                        = Additional rate band at nil rate ({0} at {1}%)
 taxCal_breakdown.table.totalIncomeTaxAndNicsDue                     = Income Tax and National Insurance contributions due
 taxCal_breakdown.table.nic4                                         = Class 4 National Insurance ({0} at {1}%)
+taxCal_breakdown.table.BRT.uplift                                   = 1.25% uplift in National Insurance contributions funds NHS, health and social care
+taxCal_breakdown.national_insurance_contributions                   = National Insurance contributions
```
Comment (chandraAndem) [inline conf/messages:237]: Maybe worth checking with design team about character **I** being capital in National Insurance but not in National insurance type.

**Review by chandraAndem — CHANGES_REQUESTED** [review]

File: app/views/TaxCalcBreakdownNew.scala.html [hunk 2]
```diff
@@ -337,6 +295,58 @@ <h1 class="govuk-heading-xl">
...
+            @viewModel.nic4Bands.map { nicBands =>
+                @for(band <- nicBands.filter(_.income > 0).sortBy(_.rate)) {
+                    <tr class="govuk-table__row">
+                        <td class="govuk-table__cell">
+                            @messages(
+                                s"taxCal_breakdown.table.nic4.${band.name}",
+                                BigDecimal(band.income).toCurrencyString,
+                                s"${band.rate}"
+                            )
+                            @if(class4UpliftEnabled && Set("BRT", "HRT").contains(band.name)) {
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxCalcBreakdownNew.scala.html]: I agree. Done, thanks.

File: app/views/TaxCalcBreakdownNew.scala.html [hunk 3]
```diff
@@ -337,6 +295,58 @@ <h1 class="govuk-heading-xl">
...
+        </tr>
+
+        @if(viewModel.nic4Bands.isDefined) {
+            <tbody class="govuk-table__body">
+            @viewModel.nic4Bands.map { nicBands =>
+                @for(band <- nicBands.filter(_.income > 0).sortBy(_.rate)) {
+                    <tr class="govuk-table__row">
+                        <td class="govuk-table__cell">
+                            @messages(
+                                s"taxCal_breakdown.table.nic4.${band.name}",
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxCalcBreakdownNew.scala.html]: You are right. Done, thanks

File: conf/messages [hunk 4]
```diff
@@ -218,18 +218,23 @@ taxCal_breakdown.additional_charges                                 = Additional
 taxCal_breakdown.table.capital_gains_tax                            = Capital Gains Tax
 taxCal_breakdown.additional_deductions                              = Additional deductions
 taxCal_breakdown.table.ZRT                                          = Zero rate ({0} at {1}%)
+taxCal_breakdown.table.nic4.ZRT                                     = Class 4 National Insurance ({0} at {1}%)
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply conf/messages]: Done, thanks

File: conf/messages [hunk 5]
```diff
@@ -218,18 +218,23 @@ taxCal_breakdown.additional_charges                                 = Additional
...
 taxCal_breakdown.table.HRT                                          = Higher rate ({0} at {1}%)
+taxCal_breakdown.table.nic4.HRT                                     = Class 4 National Insurance ({0} at {1}%)
 taxCal_breakdown.table.ART                                          = Additional rate ({0} at {1}%)
 taxCal_breakdown.table.ART_scottish                                 = Top rate ({0} at {1}%)
 taxCal_breakdown.table.ZRTBR                                        = Basic rate band at nil rate ({0} at {1}%)
 taxCal_breakdown.table.ZRTHR                                        = Higher rate band at nil rate ({0} at {1}%)
 taxCal_breakdown.table.ZRTAR                                        = Additional rate band at nil rate ({0} at {1}%)
 taxCal_breakdown.table.totalIncomeTaxAndNicsDue                     = Income Tax and National Insurance contributions due
 taxCal_breakdown.table.nic4                                         = Class 4 National Insurance ({0} at {1}%)
+taxCal_breakdown.table.BRT.uplift                                   = 1.25% uplift in National Insurance contributions funds NHS, health and social care
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply conf/messages]: Done, thanks

File: app/views/TaxCalcBreakdownNew.scala.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxCalcBreakdownNew.scala.html]: Done, thanks

File: app/views/TaxCalcBreakdownNew.scala.html [hunk 6]
```diff
@@ -337,6 +295,58 @@ <h1 class="govuk-heading-xl">
...
+                                BigDecimal(band.income).toCurrencyString,
+                                s"${band.rate}"
+                            )
+                            @if(class4UpliftEnabled && Set("BRT", "HRT").contains(band.name)) {
+                                <div>
+                                    <span class="govuk-body-s uplift">
+                                    @messages("taxCal_breakdown.table.BRT.uplift")
```
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxCalcBreakdownNew.scala.html]: Done, thanks

File: app/views/TaxCalcBreakdownNew.scala.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply app/views/TaxCalcBreakdownNew.scala.html]: You are right. Done, thanks

File: conf/messages (hunk 4 above)
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply conf/messages]: Nice one. Done, thanks

File: conf/messages (hunk 5 above)
[the remark this answers is not in the document]
Comment (andriikondiuk) [reply conf/messages]: Done, thanks

File: conf/messages (hunk 1 above)
Comment (andriikondiuk) [reply conf/messages:234]: Done, thanks

**Review by chandraAndem — APPROVED** [review]

**Pull request merged**